internal/services/network/public_ips_data_source.go

```diff
@@ -125,7 +128,7 @@ func dataSourcePublicIPsRead(d *pluginsdk.ResourceData, meta interface{}) error
 		filteredIPAddresses = append(filteredIPAddresses, element)
 	}
 
-	d.SetId(time.Now().UTC().String())
+	d.SetId(fmt.Sprintf("publicIPs/resourceGroup:%s/namePrefix:%s/attachmentStatus:%s/allocationType:%s", resourceGroup, prefix, attachmentStatus, allocationType))
```

Comment:
```suggestion
	d.SetId(fmt.Sprintf("network_publicIPs/resourceGroup/%s;namePrefix=%s;attachmentStatus=%s;allocationType=%s", resourceGroup, prefix, attachmentStatus, allocationType))
```

Reply:
this should be using a ARM Style Resource ID (or [base64] encoding this value prior to setting it into the state)